internal/pkg/stream/cloudformation.go

```diff
@@ -77,9 +80,16 @@ func (s *StackStreamer) Fetch() (next time.Time, err error) {
 			StackName: aws.String(s.stackName),
 		})
 		if err != nil {
+			// Check for throttles and wait to try again using the StackStreamer's interval.
+			s.retries += 1
+			if request.IsErrorThrottle(err) {
+				return sleep(s.retries), nil
+			}
 			return next, fmt.Errorf("describe stack events %s: %w", s.stackName, err)
 		}
 
+		s.retries = 0
```

Comment:
Maybe we want to set it to 0 before the for loop so as to make sure we start from 0.

Reply:
I think the flow goes like -
1. `s.retries` is initialized to 0 on construction.
2. If the request is a throttle, the counter is incremented for backoff. 
3. If the request is not throttled, the counter is reset to 0. 
So unless I'm missing something, this should have the correct behavior.

Reply:
For now it is but since retries is a member variable, when it is used by the other methods, I'm not sure if it will be always 0 when it is firstly used in this method. But I agree this might be just a nit tho.

Reply:
I think the intent is to not use this field outside of the Fetch method, and if we need other retriers we can add a separate field. This is currently safe because there's only ever one ongoing call of Fetch per client, and the subscribers are being updated with the results of that call.